urts/src/lib.rs

```diff
@@ -93,35 +179,77 @@ impl EnclaveBuilder {
                 self.debug as c_int,
                 &mut enclave_id,
                 ptr::null_mut(),
-                0,
-                ptr::null_mut(),
+                ex_features,
+                &mut ex_features_p as *mut *const c_void,
             )
-        };
-        match result {
-            sgx_status_t::SGX_SUCCESS => Ok(Enclave::new(enclave_id)),
-            error => Err(Error::SgxStatus(error)),
         }
+        .into_result()
+        .map_err(Error::from)
+        .map(|_| Enclave { id: enclave_id })
     }
 }
 
 impl From<Vec<u8>> for EnclaveBuilder {
-    fn from(bytes: Vec<u8>) -> EnclaveBuilder {
+    fn from(input: Vec<u8>) -> EnclaveBuilder {
         EnclaveBuilder {
-            bytes,
+            bytes: input,
             debug: false,
+            pcl_key: None,
+            kss_config: None,
         }
     }
 }
 
-impl Enclave {
-    fn new(id: sgx_enclave_id_t) -> Enclave {
-        Enclave { id }
+impl From<&[u8]> for EnclaveBuilder {
```

Comment:
I think keeping `TryFrom<File>` is worth this small repetition. I did reorganize the `From` impls so that they all call the `From<Vec<u8>>`, since everything needs to be turned into a vec at the bottom anyway.